ts/src/kraken.ts

```diff
@@ -2881,44 +2891,23 @@ export default class kraken extends Exchange {
     }
 
     handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
-        if (code === 520) {
-            throw new ExchangeNotAvailable (this.id + ' ' + code.toString () + ' ' + reason);
-        }
-        // todo: rewrite this for "broad" exceptions matching
-        if (body.indexOf ('Invalid order') >= 0) {
-            throw new InvalidOrder (this.id + ' ' + body);
-        }
-        if (body.indexOf ('Invalid nonce') >= 0) {
-            throw new InvalidNonce (this.id + ' ' + body);
-        }
-        if (body.indexOf ('Insufficient funds') >= 0) {
-            throw new InsufficientFunds (this.id + ' ' + body);
-        }
-        if (body.indexOf ('Cancel pending') >= 0) {
-            throw new CancelPending (this.id + ' ' + body);
-        }
-        if (body.indexOf ('Invalid arguments:volume') >= 0) {
-            throw new InvalidOrder (this.id + ' ' + body);
-        }
-        if (body.indexOf ('Rate limit exceeded') >= 0) {
-            throw new RateLimitExceeded (this.id + ' ' + body);
-        }
-        if (response === undefined) {
-            return undefined;
-        }
```

Comment:
Why did you decide to remove this mapping? You could just replace `body` with `reason`

Reply:
I moved the mapped strings to ....exceptions.broad ...
Because it's the way it is supposed to be in ccxt. And then I used
`
this.throwExactlyMatchedException (this.exceptions['exact'], error, 
this.throwBroadlyMatchedException (this.exceptions['broad'], error, error);
`
To have the exact same mapping working